core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -780,7 +780,9 @@ private void handleRequestOrResponseComplete() {
                         // Stop receiving new requests.
                         handledLastRequest = true;
                         if (unfinishedRequests.isEmpty()) {
-                            ctx.writeAndFlush(Unpooled.EMPTY_BUFFER).addListener(CLOSE);
+                            ctx.executor().schedule(() ->
+                                            ctx.writeAndFlush(Unpooled.EMPTY_BUFFER).addListener(CLOSE),
+                                            1000, TimeUnit.MILLISECONDS);
```

Comment:
optional nit; using seconds is probably slightly easier to reason about

```suggestion
                                            1, TimeUnit.SECONDS);
```

Reply:
I think this value can be configured in milliseconds on fast networks.